purchases/src/main/kotlin/com/revenuecat/purchases/BillingWrapper.kt

```diff
@@ -169,6 +186,9 @@ internal class BillingWrapper internal constructor(
         } else {
             debugLog("Making purchase for sku: ${skuDetails.sku}")
         }
+        synchronized(this@BillingWrapper) {
+            productTypes[skuDetails.sku] = skuDetails.type
+        }
```

Comment:
Are there risks to trusting the caller as the to the type of a SKU, rather than relying on our own fetching?

Reply:
That's a good point. Maybe it's time to remove this deprecated method